lib/src/config/config.rs

```diff
@@ -377,13 +376,26 @@ impl Config {
     /// # Ok(())
     /// # }
     /// ```
-    pub fn set_address<A: Into<String>>(&mut self, address: A) -> Result<()> {
-        let address = address.into();
-        if (&*address, 0u16).to_socket_addrs().is_err() {
+    pub fn set_address<A: TryInto<Address>>(&mut self, address: A) -> Result<()> {
```

Comment:
This looks really complicated. Why are we using `try_into()` when `parse()` would work just as well?

I'd still like to abstract all of this away into one module with a big `#[cfg(unix)]` and `#[cfg(not(unix))]`. I don't like having these `cfg!` calls sprinkled everywhere; it makes the code much harder to follow.